pkg/cmd/extension/manager.go

```diff
@@ -802,3 +825,23 @@ func codesignBinary(binPath string) error {
 	cmd := exec.Command(codesignExe, "--sign", "-", "--force", "--preserve-metadata=entitlements,requirements,flags,runtime", binPath)
 	return cmd.Run()
 }
+
+// cleanExtensionUpdateDir should be used before installing extensions to avoid past metadata creating problems.
+func (m *Manager) cleanExtensionUpdateDir(name string) error {
+	updatePath := m.UpdateDir(name)
+	if _, err := os.Stat(updatePath); err == nil {
```

Comment:
Consider checking for error type being a file that does not exist specifically. 

Example:

https://github.com/cli/cli/blob/b3f74900f2f078e5fa52483775ed316595a589a3/pkg/cmd/extension/manager.go#L214-L226